packages/dds/tree/src/test/utils.ts

```diff
@@ -823,16 +829,29 @@ function createCheckoutWithContent(
 			IEmitter<CheckoutEvents> &
 			HasListeners<CheckoutEvents>;
 		additionalAsserts?: boolean;
+		// Creates a chunked forest instead of an object forest.
+		chunkedForest?: boolean;
 	},
 ): { checkout: TreeCheckout; logger: IMockLoggerExt } {
 	const fieldCursor = normalizeNewFieldContent(content.initialTree);
-	const roots: MapTree = mapTreeWithField(mapTreeFieldFromCursor(fieldCursor));
 	const schema = new TreeStoredSchemaRepository(content.schema);
-	const forest = buildTestForest({
-		additionalAsserts: args?.additionalAsserts ?? true,
-		schema,
-		roots,
-	});
+	let forest: IEditableForest;
+	if (args?.chunkedForest === true) {
+		forest = buildConfiguredForest(
+			new Breakable("buildTestChunkedForest"),
+			ForestTypeOptimized,
+			schema,
+			testIdCompressor,
+		);
+		initializeForest(forest, fieldCursor, testRevisionTagCodec, testIdCompressor);
+	} else {
+		const roots: MapTree = mapTreeWithField(mapTreeFieldFromCursor(fieldCursor));
+		forest = buildTestForest({
+			additionalAsserts: args?.additionalAsserts ?? true,
+			schema,
+			roots,
+		});
+	}
```

Comment:
I think we can remove this branch of the iff and make everything work the same way by using `ForestType`

ForestTypeExpensiveDebug is an object forest with additional asserts enabled, ForestTypeReference in the object forest without addational asserts, and ForestTypeOptimized is the ChunkedForest.

Thus checkoutWithContent could take in a ForestTyoe rather than the additionalAsserts and chunkedForest flags.

I think that would be cleaner, and would provide a way for any callers who need it in the future to do custom configured forests (with a custom ForestType)

Reply:
I tried this to start with and tests started failing. I tried again after your comment and they are working fine! :D